src/Makefile

```diff
@@ -436,13 +435,17 @@ DEP = $(REDIS_SERVER_OBJ:%.o=%.d) $(REDIS_CLI_OBJ:%.o=%.d) $(REDIS_BENCHMARK_OBJ
 %.o: %.c .make-prerequisites
 	$(REDIS_CC) -MMD -o $@ -c $<
 
-# The file commands.c is checked in and doesn't normally need to be rebuilt. It
+# The file commands.def is checked in and doesn't normally need to be rebuilt. It
 # is built only if python is available and its prereqs are modified.
 ifneq (,$(PYTHON))
-$(COMMANDS_FILENAME).c: commands/*.json ../utils/generate-command-code.py
+$(COMMANDS_DEF_FILENAME).def: commands/*.json ../utils/generate-command-code.py
 	$(QUIET_GEN)$(PYTHON) ../utils/generate-command-code.py $(GEN_COMMANDS_FLAGS)
 endif
 
+commands.c: $(COMMANDS_DEF_FILENAME).def
+
+cli_commands.c: commands.def
+
```

Comment:
See https://github.com/redis/redis/pull/10515#discussion_r1151677470

Reply:
> The `.c` files are not generated anymore. Only the `.o` files depend on the `.c` files and the `.def` files. Shouldn't we have rules like `commands.o: commands.c commands.def` instead, or am I missing something?

i'm not that good with makefiles, maybe i'm missing something.
i think we have a generic rule for `o` -> `c` dependency. so we don't need to state that commands.o depends on commands.c

we also have some mechanism for automatic dependencies mechanism, maybe it can detect the inclusion of commands.def by itself.

feel free to add commits to improve / clean it (although maybe we should wait to be sure we wanna proceed with this and not revert it)

Reply:
right. i guess i was too lazy to check, i'll remove them.